Unit tests for admission controllers flaky on Travis

Here is a sample of a flake on an admission controller.

```
--- FAIL: TestWithNoPlanWorksWithSinglePlan (0.00s)
	admission_test.go:212: unexpected error "serviceinstances.servicecatalog.k8s.io \"instance\" is forbidden: no plans found at all for service class \"foo\"" returned from admission handler: list:serviceclasses:, watch:serviceclasses:, 
	admission_test.go:216: PlanName was not modified for the default plan
```

--- a/plugin/pkg/admission/serviceplan/changevalidator/admission.go
+++ b/plugin/pkg/admission/serviceplan/changevalidator/admission.go
@@ -121,7 +121,12 @@ func (d *denyPlanChangeIfNotUpdatable) SetInternalServiceCatalogInformerFactory(
 	d.scLister = scInformer.Lister()
 	spInformer := f.Servicecatalog().InternalVersion().ServicePlans()
 	d.spLister = spInformer.Lister()
-	d.SetReadyFunc(scInformer.Informer().HasSynced)
+
+	readyFunc := func() bool {
+		return scInformer.Informer().HasSynced() && instanceInformer.Informer().HasSynced() && spInformer.Informer().HasSynced()
+	}
+
+	d.SetReadyFunc(readyFunc)
 }
 
 func (d *denyPlanChangeIfNotUpdatable) Validate() error {

--- a/plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go
+++ b/plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go
@@ -144,7 +144,7 @@ func (d *defaultServicePlan) SetInternalServiceCatalogInformerFactory(f informer
 	d.spLister = spInformer.Lister()
 
 	readyFunc := func() bool {
-		return scInformer.Informer().HasSynced() && scInformer.Informer().HasSynced()
+		return scInformer.Informer().HasSynced() && spInformer.Informer().HasSynced()
 	}
 
 	d.SetReadyFunc(readyFunc)

